src/core/symbology/qgsstyle.cpp

```diff
@@ -2803,13 +2803,13 @@ bool QgsStyle::importXml( const QString &filename, int sinceVersion )
           favorite = true;
         }
 
-        QgsSymbol *symbol = QgsSymbolLayerUtils::loadSymbol( e, QgsReadWriteContext() );
+        std::unique_ptr< QgsSymbol > symbol = QgsSymbolLayerUtils::loadSymbol( e, QgsReadWriteContext() );
         if ( symbol )
         {
-          addSymbol( name, symbol );
+          addSymbol( name, symbol->clone() );
```

Comment:
why not just release here?

Reply:
It's used a couple of lines later

Reply:
Yes, but could we not saveSymbol() before and then release to avoid the creation of a new object during clone?

or store it in a raw pointer if we want to keep the instruction order

Reply:
OK, I've gone with the raw pointer option -- saveSymbol fires a bunch of signals, and I'm not confident that listeners won't be affected if we haven't called addSymbol yet.